core/stat/resource_node.go

```diff
@@ -45,3 +45,7 @@ func (n *ResourceNode) ResourceType() base.ResourceType {
 func (n *ResourceNode) ResourceName() string {
 	return n.resourceName
 }
+
+func (n *ResourceNode) RealBucketLeapArray() *sbase.BucketLeapArray {
```

Comment:
Maybe this is not necessary? The inner LeapArray should be invisible to users.

Reply:
It is used in circuit breaker.
```go
resNode := stat.GetOrCreateResourceNode(rule.ResourceName())
bla := resNode.RealBucketLeapArray()
```
when we new SlidingWindowMetric, we need pass parameter "*BucketLeapArray".

Reply:
There may be other ways to implement this requirement.

Reply:
It's okay for now. Some design needs to be refactored later.